File: PAT/初级/1003.cpp
#include <cstdio>
#include <cstring>

const int maxn = 105;
char s[maxn];

bool judge() {
    bool hasP = false;
    bool hasT = false;
    bool wrong = false;
    int aNums[3] = {0, 0, 0};
    int len = strlen(s);
    for (int i = 0; i < len && !wrong; ++i) {
        if (s[i] == 'A') {
            if (!hasP && !hasT)
                ++aNums[0];
            if (hasP && !hasT)
                ++aNums[1];
            if (hasP && hasT)
                ++aNums[2];
            if (!hasP && hasT)
                wrong = true;
            continue;
        }
        if (s[i] == 'P') {
            if (hasP)
                wrong = true;
            else
                hasP = true;
            continue;
        }
        if (s[i] == 'T') {
            if (hasT)
                wrong = true;
            else
                hasT = true;
            continue;
        }
        wrong = true;
    }
    return hasP && hasT && aNums[1] >= 1 && (aNums[1] * aNums[0] == aNums[2]) &&
           !wrong;
}

int main() {
    int n;
    scanf("%d", &n);
    for (int i = 0; i < n; ++i) {
        scanf("%s", s);
        printf("%s\n", judge() ? "YES" : "NO");
    }
    return 0;
}

File: PAT/初级/1003.py
import re


def judge(s):
    wrong = False
    l = s.split('P')
    if len(l) != 2:
        return False
    r = l[1].split('T')
    if len(r) != 2:
        return False
    if not re.match(r'[PAT]+', s):
        return False

    # print([l[0], r[0], r[1]])

    return (len(l[0]) * len(r[0]) == len(r[1])) and len(r[0]) > 0


n = int(input())
for i in range(n):
    s = input()
    print("YES" if judge(s) else "NO")
